blog/views.py

```diff
@@ -2,21 +2,36 @@
 from django.views.generic import ListView, DetailView, View
 from .models import Post
 from episodes.models import Category, Tags
+from rest_framework import viewsets
+
 # Create your views here.
+from .serializers import PostSerializer
+
+
 class BlogListView(ListView):
     model = Post
     template_name = 'blog/blog_list.html'
+
     def get_context_data(self, **kwargs):
         post_list = Post.objects.filter(status='published')
         context = super().get_context_data(**kwargs)
-        context['post_list']=post_list
+        context['post_list'] = post_list
         context['categories'] = Category.objects.all()
-        context['tags'] = Tags.objects.all() 
+        context['tags'] = Tags.objects.all()
         return context
+
+
 class BlogDetailView(View):
     model = Post
+
     def get(self, request, detail_post_slug, *args, **kwargs):
         post = get_object_or_404(Post, slug=detail_post_slug)
         # gallery = InterviewImage.objects.filter(interview__title=interview)
         context = {'post_details': post}
-        return render(request, "blog/blog_details.html", context)
\ No newline at end of file
+        return render(request, "blog/blog_details.html", context)
+
+
+class PostViewSet(viewsets.ModelViewSet):
```

Comment:
This would be Post (blogpost) ListView which is already there (BlogListView)
http_method_names are not necessary because the 4 methods we operate are GET POST PUT DELETE
which are implemented as class' methods - next comment